vsintegration/src/FSharp.VS.FSI/fsiSessionToolWindow.fs

```diff
@@ -530,19 +530,31 @@ type internal FsiToolWindow() as this =
         try
             showNoActivate()
             let directiveC  = sprintf "# 1 \"stdin\""    (* stdin line number reset code *)                
-            let text = "\n" + text + "\n" + directiveC + "\n;;\n"
+            let text() = "\n" + text + "\n" + directiveC + "\n;;\n"
             executeTextNoHistory null text
         with _ -> ()
 
+    let hide () =
+        if sessions.SupportsInteractivePrompt then
+            """#interactiveprompt "hide" """
+        else
+            ""
+
+    let show () =
+        if sessions.SupportsInteractivePrompt then
+            """#interactiveprompt "show" """
+        else
+            ""
+
     let executeInteraction dbgBreak dir filename topLine (text:string) =
-        let interaction = $"""
-#interactiveprompt "hide"
+        let interaction() = $"""
```

Comment:
```suggestion
        let interaction = $"""
```

Reply:
For here and all the suggestions above:

The text was represented as a lazy string (unit -> string). but in both usages passed in for execution immediately.

Keeping it as a simple string works the same, unless I missed some aspect of laziness being exercised here.

Reply:
> For here and all the suggestions above:
> 
> The text was represented as a lazy string (unit -> string). but in both usages passed in for execution immediately.
> 
> Keeping it as a simple string works the same, unless I missed some aspect of laziness being exercised here.

I'm afraid it is a function call. we need to delay getting the value for SupportsInteractivePrompt until after we have started fsi.exe, otherwise the string may be composed before fsi has been started.

Reply:
That functional call is doing interpolation of existing fragments, I do not see anything which is getting delayed here.

Or you mean the transitive usage of sessions.SupportsInteractivePrompt has to be delayed, because that value will change>?